docs/madr/decisions/038-transition-to-new-policies.md

```diff
@@ -0,0 +1,473 @@
+- value: 0s
+  op: replace
+  path: envoy.config.listener.v3.Listener/outbound:10.43.138.50:8080/filterChains/0/filters/0/typedConfig/commonHttpProtocolOptions/idleTimeout
+- value: 0s
+  op: add
+  path: envoy.config.listener.v3.Listener/outbound:10.43.138.50:8080/filterChains/0/filters/0/typedConfig/requestHeadersTimeout
+- value: 25s
+  op: add
+  path: envoy.config.listener.v3.Listener/outbound:10.43.138.50:8080/filterChains/0/filters/0/typedConfig/routeConfig/virtualHosts/0/routes/0/route/idleTimeout
+```
+
+Let's review these changes:
+
+* timeouts `requestHeadersTimeout` and `maxConnectionDuration` are disabled when equal to `0s` or unset. There was no 
+way to specify them with old Timeout policy.
+* `commonHttpProtocolOptions/idleTimeout` with MeshTimeout is set on cluster and disabled on the listener 
+* `route/idleTimeout` is duplicated value of `streamIdleTimeout` but per-route (previously we've set it only per-listener)
+
+### CircuitBreaker -> MeshCircuitBreaker
+
+There is an old CircuitBreaker policy:
+```yaml
+type: CircuitBreaker
+mesh: default
+name: circuit-breaker-example
+sources:
+- match:
+    kuma.io/service: "*"
+destinations:
+- match:
+    kuma.io/service: "*"
+conf:
+  interval: 21s
+  baseEjectionTime: 22s
+  maxEjectionPercent: 23
+  splitExternalAndLocalErrors: false
+  thresholds:
+    maxConnections: 24
+    maxPendingRequests: 25
+    maxRequests: 26
+    maxRetries: 27
+  detectors:
+    totalErrors: 
+      consecutive: 28
+    gatewayErrors: 
+      consecutive: 29
+    localErrors: 
+      consecutive: 30
+    standardDeviation:
+      requestVolume: 31
+      minimumHosts: 32
+      factor: 1.33
+    failure:
+      requestVolume: 34
+      minimumHosts: 35
+      threshold: 36
+```
+
+User creates a new MeshCircuitBreaker replacement:
+
+```yaml
+type: MeshCircuitBreaker
+name: backend-inbound-outlier-detection
+mesh: default
+spec:
+  targetRef:
+    kind: Mesh
+  to:
+  - targetRef:
+      kind: Mesh
+    default:
+      connectionLimits:
+        maxConnections: 24
+        maxPendingRequests: 25
+        maxRequests: 26
+        maxRetries: 27
+      outlierDetection:
+        interval: 21s
+        baseEjectionTime: 22s
+        maxEjectionPercent: 23
+        splitExternalAndLocalErrors: false
+        detectors:
+          totalFailures:
+            consecutive: 28
+          gatewayFailures:
+            consecutive: 29
+          localOriginFailures:
+            consecutive: 30
+          successRate:
+            requestVolume: 31
+            minimumHosts: 32
+            standardDeviationFactor: 1.33
+          failurePercentage:
+            requestVolume: 34
+            minimumHosts: 35
+            threshold: 36
+```
+
+Using shadow mode we see there is an empty jsonPatches list!
+
+### FaultInjection -> MeshFaultInjection
+
+It should be always safe to remove FaultInjection and create new MeshFaultInjection when necessary. 
+
+### TrafficPermission -> MeshTrafficPermission
```

Comment:
Do we need to provide more explanation for MeshTrafficPermission since in 2.6 it creates HTTP rbac configuration?

Reply:
It's pretty bad even with TCP rbac, but with HTTP the diff looks even worse:
```yaml
- value:
    - name: envoy.filters.network.http_connection_manager
      typedConfig:
        '@type': type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
        commonHttpProtocolOptions:
          idleTimeout: 0s
        forwardClientCertDetails: SANITIZE_SET
        httpFilters:
          - name: envoy.filters.http.rbac
            typedConfig:
              '@type': type.googleapis.com/envoy.extensions.filters.http.rbac.v3.RBAC
              rules:
                policies:
                  MeshTrafficPermission:
                    permissions:
                      - any: true
                    principals:
                      - any: true
          - name: envoy.filters.http.router
            typedConfig:
              '@type': type.googleapis.com/envoy.extensions.filters.http.router.v3.Router
        requestHeadersTimeout: 0s
        routeConfig:
          name: inbound:frontend_kuma-demo_svc_8080
          requestHeadersToRemove:
            - x-kuma-tags
          validateClusters: false
          virtualHosts:
            - domains:
                - '*'
              name: frontend_kuma-demo_svc_8080
              routes:
                - match:
                    prefix: /
                  route:
                    cluster: localhost:8080
                    idleTimeout: 3600s
                    timeout: 0s
        setCurrentClientCertDetails:
          uri: true
        statPrefix: localhost_8080
        streamIdleTimeout: 3600s
  op: replace
  path: /type.googleapis.com~1envoy.config.listener.v3.Listener/inbound:10.42.0.16:8080/filterChains/0/filters
```
I'll add this to the doc